core/src/main/java/novoda/lib/sqliteprovider/analyzer/ForeignTablesQuery.java

```diff
@@ -0,0 +1,41 @@
+package novoda.lib.sqliteprovider.analyzer;
+
+import android.database.Cursor;
+
+import java.util.List;
+
+class ForeignTablesQuery implements Query<String> {
+
+    private final String table;
+    private final List<String> allDatabaseTables;
+
+    public ForeignTablesQuery(String table, List<String> allDatabaseTables) {
+        this.table = table;
+        this.allDatabaseTables = allDatabaseTables;
+    }
+
+    @Override
+    public String getSqlStatement() {
+        return "PRAGMA table_info(\"" + table + "\");";
+    }
+
+    @Override
+    public String parseRow(Cursor cursor) {
+        String name = getTableName(cursor);
```

Comment:
So `getTableName(cursor)` returns a name, but we have a new `tableName` in the next lines of code, what is what? :dancer:

As well we have an `if` inside another `if`.